packages/angular-material/src/other/master-detail/master.ts

```diff
@@ -247,6 +249,15 @@ export class MasterListComponent extends JsonFormsArrayControl {
   }
 }
 
+const getFirstUsableProperty = (schema: JsonSchema): string => {
+  const prop = Object.entries(schema?.properties).find(e => {
+    if(e[1].type === 'object' || e[1].properties) return false;
+    if(e[1].type === 'array' || e[1].items) return false;
+    return true;
+  });
+  return prop ? prop[0] : Object.keys(schema?.properties)[0];
+}
```

Comment:
There is a utility for this in `@jsonforms/core` which you can import: `getFirstPrimitiveProp`.